pkg/gofr/http/responder.go

```diff
@@ -53,9 +53,18 @@ func (r Responder) Respond(data any, err error) {
 		r.w.Header().Set("Content-Type", "application/json")
 	}
 
-	r.w.WriteHeader(statusCode)
+	jsonData, encodeErr := json.Marshal(resp)
+	if encodeErr != nil {
+		r.w.WriteHeader(http.StatusInternalServerError)
+
+		_, _ = r.w.Write([]byte(`{"error":{"message": "failed to encode response as JSON"}}` + "\n"))
+
+		return
+	}
 
-	_ = json.NewEncoder(r.w).Encode(resp)
+	r.w.WriteHeader(statusCode)
+	_, _ = r.w.Write(jsonData)
```

Comment:
It's a bit out of the scope of the PR, but do you have a way like a logger to report you got an error when writting to output.

Usually, it's a connection closed issue, but it could be interesting to catch them